model/telemetry/registry.yaml

```diff
@@ -81,3 +81,57 @@ groups:
         brief: >
           The version string of the auto instrumentation agent or distribution, if used.
         examples: ["1.2.3"]
+      - id: telemetry.sdk.processor.type
+        type:
+          members:
+            - id: batch_span
+              value: "batch-span"
+              brief: >
+                The builtin SDK Batching Span Processor
+              stability: experimental
+            - id: simple_span
+              value: "simple-span"
+              brief: >
+                The builtin SDK Simple Span Processor
+              stability: experimental
+        stability: experimental
+        brief: >
+          A name identifying the type of the OpenTelemetry SDK processor.
+        examples: ["batch-span", "MyCustomProcessor"]
+      - id: telemetry.sdk.exporter.type
+        type:
+          members:
+            - id: otlp_grpc
+              value: "otlp-grpc"
+              brief: >
+                OTLP exporter over gRPC with protobuf serialization
+              stability: experimental
+            - id: otlp_http
+              value: "otlp-http"
+              brief: >
+                OTLP exporter over HTTP with protobuf serialization
+              stability: experimental
+            - id: otlp_http_json
+              value: "otlp-http-json"
+              brief: >
+                OTLP exporter over HTTP with JSON serialization
+              stability: experimental
+        stability: experimental
+        brief: >
+          A name identifying the type of the OpenTelemetry SDK exporter.
+        examples: ["otlp-grpc", "jaeger"]
+      - id: telemetry.sdk.component.id
+        type: string
+        stability: experimental
+        brief: >
+          A name uniquely identifying the instance of the OpenTelemetry SDK component within its containing SDK instance.
```

Comment:
Is this necessary? It does not map to anything standard in the otel SDK and we can capture the same level of details if we captured the component name (e.g. class/type name) in the instrumentation scope name or had one attribute for component name and `telemetry.sdk.exporter|processor.type`.

E.g. `telemetry.sdk.component.name` would  contain a fully qualified name of the processor or exporter such as `io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter`.

Reply:
It is required to ensure the uniqueness of individual time series for gauges and updown-counters:

For example `telemetry.sdk.span.processor.queue_capacity`: The SDK explicitly allows you to setup any amount of span processors you like: For example, you can set up two `BatchSpanProcessor`s, each exporting to a different backend. In that case `queue_capacity` would break, because there is no distinguishing attribute for the two timeseries (one for each processor instance). This is fixed by adding the `telemetry.sdk.component.id`.

I agree that 98% case is having just one processor instance per type, but I still want the metrics to remain functional in the 2% case.

And yes, there is currently no concept for naming components explicitly in the SDK and I don't expect any to be there soon. With the definition here I just wanted to be the least prescriptive and leave the door open for adding such a naming mechanism in the future.